control/src/tcp/attributes_builder.h

```diff
@@ -27,16 +27,24 @@ namespace tcp {
 // The builder class to add TCP attributes.
 class AttributesBuilder {
  public:
-  AttributesBuilder(RequestContext* request) : request_(request) {}
+  AttributesBuilder(RequestContext* request)
+      : request_(request),
+        last_report_info_{0ULL, 0ULL, std::chrono::nanoseconds::zero()} {}
 
   // Extract attributes for Check.
   void ExtractCheckAttributes(CheckData* check_data);
   // Extract attributes for Report.
-  void ExtractReportAttributes(ReportData* report_data);
+  void ExtractReportAttributes(ReportData* report_data, bool is_final_report);
 
  private:
   // The request context object.
   RequestContext* request_;
+
+  // Records reported information in last Report() call. This is needed for
+  // calculating delta information which will be sent in periodical report.
+  // Delta information includes incremented sent bytes and received bytes
+  // between last report and this report.
+  ReportData::ReportInfo last_report_info_;
```

Comment:
I prefer not to use AttributesBuilder to store "last" info.  "last" should be stored in request_impl.

AttributeBuilder is one shot think, as it always allocated from stack as

Builder b(&attributes)
b.Setxxxx